cmake/CMakeLists.txt

```diff
@@ -470,7 +470,10 @@ if (onnxruntime_USE_JEMALLOC)
   list(APPEND onnxruntime_EXTERNAL_DEPENDENCIES jemalloc)
 endif()
 
-include_directories(${ONNXRUNTIME_INCLUDE_DIR})
+include_directories(
+  ${ONNXRUNTIME_INCLUDE_DIR} 
+  ${REPO_ROOT}/include/onnxruntime/core/session
```

Comment:
Why this change is needed?

Reply:
Unless we refactor the provider_factory headers, they need to include the onnxruntime_c_api.h. we don't want to export all the path structure in the package, rather put header files under the root. So they need to include the c_api.h without path. To get the current build working, we thus need to put the core/session in the include path.

Cleaner option would be to completely separate out the c_api.h and provider_factory headers in the root of the include similar to what we export. Would defer that to some change that refactors the providers.

Reply:
This is a hack. It's ok for now, but we should fix it at a later time.